lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -109,6 +113,74 @@ def get(self, request, course_id):
         return Response(get_course(request, course_key))
 
 
+class CourseUsersView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Fetch statistics about a user's activity in a course.
+
+    **Example Requests**:
+
+        GET /api/discussion/v1/users/?course_id=course-v1:ExampleX+Subject101+2015&order_by=activity
+
+    **Response Values**:
+
+
+    **Example Response**
+    ```json
+    {
+        "pagination": {
+            "count": 3,
+            "next": null,
+            "num_pages": 1,
+            "previous": null
+        },
+        "results": [
+            {
+                "active_flags": 3,
+                "inactive_flags": 0,
+                "replies": 13,
+                "responses": 21,
+                "threads": 32,
+                "username": "edx"
+            },
+            {
+                "active_flags": 1,
+                "inactive_flags": 0,
+                "replies": 6,
+                "responses": 8,
+                "threads": 13,
+                "username": "honor"
+            },
+            ...
+        ]
+    }
+    ```
+
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser)
+
+    def get(self, request):
+        """Implements the GET method as described in the class docstring."""
+        form = UserCourseStatsForm(request.query_params)
+        if not form.is_valid():
+            raise ValidationError(form.errors)
+        data = get_course_discussion_user_stats(
```

Comment:
given that the api now adds user stats, how we are paginating the results in the fronted? are we displaying all data without pagination? if we are showing data in chunks/pages, do we fetch all the data at once from the backend?

Reply:
No, the data is returned paginated from the backend, and the frontend needs to explicitly request the next page.

Reply:
got it, can you let me know the code where the pagination is being performed? I am not able to find where the data is being paginated. Thanks.